### ISSUE TITLE
[Feature] Button to clear [Log]

### ISSUE THREAD

**Issue by xehnlp** [NPBruce/valkyrie#1583]
When playing long games or campaign games the huge amount of text lines in the save.ini file will start to lag out the whole application, this is more noticable when playing on an android device. Manually clearing the [Log] section of the save.ini file fixes the lag. Having a way to clear the log inside the application would be a huge improvement to avoid this problem.

**Comment by xehnlp** [comment]
Attaching example file of a save.ini containing over 5k lines of log entries.

[save2.zip](https://github.com/NPBruce/valkyrie/files/9645026/save2.zip)

**Comment by Quantumrunner** [comment]
Very good finding! This seems also to be the root cause of this issue I had created some time ago: #1484

**Comment by mayjak** [comment]
Ok, after a brief investigation it seems that the slowness comes from the Undo feature. Essentially the whole log is tuned to a string on pretty much every button press. I have a few ideas on improvements.

**Comment by mayjak** [comment]
I added a solution that doesn't involve clearing the log (for debugging purposes if something goes wrong at a later point in the campaign) or manual actions. 
I also added an additional "Undo" notice to the logs.

Technically:
1. ArrayList moved to LinkedList (we never access the log randomly anyway).
2. The logs are ignored while serializing the undo state and the current's Quest log is used instead. This gives us an additional benefit of having an "Undo" notice in the logs.

To be released in 2.5.8 this week.
